packages/flame_3d/lib/src/extensions/quaternion.dart

```diff
@@ -0,0 +1,64 @@
+import 'dart:math';
+
+import 'package:flame_3d/core.dart';
+
+extension QuaternionExtension on Quaternion {
+  List<double> get storage => [x, y, z, w];
+
+  double dot(Quaternion other) {
+    return x * other.x + y * other.y + z * other.z + w * other.w;
+  }
+
+  static Quaternion slerp(
```

Comment:
I was really over the fence about it, because it kinda loses some beautiful symmetry when we do it like that, while at the same time I also agree it is nicer to invoke. Since you prefer that route, that is enough to sway me that direction, so I will change

Reply:
ended up keeping both options, I think both are valid and read well in certain scenarios

Reply:
Will there not be a name collision when both are used in the same file? 🤔

Reply:
No because they are on different "classes" - see the test file, you can use both ways np